src/app/functions/server/featureFlags.test.ts

```diff
@@ -0,0 +1,58 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+// import { it, expect, jest } from "@jest/globals";
+
+import { getFeatureFlagByName } from "../../../db/tables/featureFlags";
+jest.mock("../../../db/tables/featureFlags");
+
+jest.mock("next-auth", () => ({
+  getServerSession: jest.fn(() => {
+    return { user: { email: "test@example.com" } };
+  }),
+}));
+
+jest.mock("../../api/utils/auth", () => ({
+  authOptions: jest.fn(),
+}));
+
+jest.mock("../../../db/tables/featureFlags", () => ({
+  getFeatureFlagByName: jest.fn(),
+}));
+
+import { isFlagEnabled } from "./featureFlags";
+
+beforeEach(() => {
+  jest.resetModules();
+});
+
+it("throws if there is no valid session", async () => {
+  await expect(isFlagEnabled("testFlag")).resolves.toBe(false);
```

Comment:
I suggested elsewhere to take the user object as an argument, but as an FYI, this test description doesn't seem to match what it does. The current implementation is actually the " or nonexistent" part of the next test.

Checking correct behaviour for no session will be easier if it's passed in as an argument, but as an FYI, this is how you could've done it:

```suggestion
it("throws if there is no valid session", async () => {
  const nextAuthMock = jest.requireMock("next-auth");
  nextAuthMock.getServerSession.mockImplementationOnce(() => undefined);
  const { isFlagEnabled } = await import("./featureFlags");
  await expect(() => isFlagEnabled("testFlag")).rejects.toThrow("No session");
```

Unfortunately after mocking a module you have to import the module that depends on it again.